### PR TITLE
[Enhancement] support removing limit operator point query

### PR DESCRIPTION
**Pull request by Jay-ju**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/ShortCircuitPlanner.java [hunk 1]
```diff
@@ -140,13 +167,27 @@ public Boolean visitLogicalProject(OptExpression optExpression, Void context) {
             return visitChild(optExpression, context);
         }
 
+        @Override
+        public Boolean visitLogicalLimit(OptExpression optExpression, Void context) {
+            if (optExpression.getOp().getOpType().equals(OperatorType.LOGICAL_LIMIT) &&
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/ShortCircuitPlanner.java:172]: the second condition is meaningless. Limit can only have one child.

File: fe/fe-core/src/main/java/com/starrocks/sql/ShortCircuitPlanner.java [hunk 2]
```diff
@@ -158,7 +199,9 @@ protected static boolean isPointScan(Table table, List<String> keyColumns, List<
                     PartitionColumnFilter filter = filters.get(keyColumn);
                     if (filter.getInPredicateLiterals() != null) {
                         cardinality *= filter.getInPredicateLiterals().size();
-                        if (cardinality > MAX_RETURN_ROWS) {
+                        // TODO(limit operator place fe)
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/ShortCircuitPlanner.java:202]: what‘s the meaning of this comment?

**Review by decster — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/ShortCircuitPlanner.java (hunk 2 above)
Comment (Jay-ju) [reply fe/fe-core/src/main/java/com/starrocks/sql/ShortCircuitPlanner.java:202]: if limit compute node can be placed fe, we can treate fe as sink node. currently we only process return 1 record .  in other words, if having limit, cardinality must be 1

File: fe/fe-core/src/main/java/com/starrocks/sql/ShortCircuitPlanner.java (hunk 1 above)
Comment (Jay-ju) [reply fe/fe-core/src/main/java/com/starrocks/sql/ShortCircuitPlanner.java:172]: ok

**Review by decster — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Pull request merged**